service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/twin/TwinProperties.java

```diff
@@ -130,18 +130,10 @@
         if (desired != null)
         {
             this.desired = TwinCollection.createFromRawCollection(desired);
-            if (desired.getVersion() != null)
-            {
-                this.desired.setVersion(desired.getVersion());
-            }
```

Comment:
The above line already does this, so no need to manually assign the version here, too